### DISCUSSION TITLE
Auth desktop (Windows) default timeout needed

### DISCUSSION THREAD

**Discussion by semyl** [ente-io/ente#2652] [Enhancements]
I moved over from Authy. That has a timeout so the app goes back to the lock screen after a certain point. I notice Ente Auth codes stay permanently visible. I want to be able to trust that if I copy a code and get on with what I am doing, that it is going to time out in the background, without me needing to specifically close the app.

**Comment by JoelAzaria** [comment]
If this is implemented, please make it an option.   
Personally, I'd be highly annoyed having to authenticate every time I needed a TOTP (I need many throughout the day).

When I want to secure my apps/desktop I lock my desktop.   Win+L for the win :)

**Reply by semyl** [reply]
There's a different degree of security you (or I) would want for a 2FA app versus a computer generally. That's why things have passwords and 2FA in the first place. While I do lock my desktop as you describe (and Windows ALSO has an auto screenlock if it is idle for too long), there are times when someone in the household (or visiting) may need to use the computer, or I walked away to my kitchen and don't lock the screen, or maybe I'm even watching something on the device and am not going to lock it then.

In those cases, you'd want the apps that require more security to time out after a certain time of not being used, just like password managers do. Because I agree with you that I would not want it to be so immediate that you are having to reauthenticate every time you need it, but at the same time if I'm not needing it for more than say 30 mins or 2 hours, I'd love to trust that it has auto timed-out. I don't like coming back to my desktop the next day(s), logging back in to Windows (I don't shut down) and realising I left Auth open and could still go about getting my 2FA codes without logging in to Auth again.

But yes, options would be ideal, with different times (5mins, 10mins, 30mins, 1hr, 4hrs, 8hrs, 12hrs, 24hrs, etc) and/or triggered also by Windows Lock Screen, if such a thing can be done.

**Comment by JoelAzaria** [comment]
Related to / Dup request #3505

**Comment by BrianG61UK** [comment]
You can configure Ente Auth to hide all the codes. Then single clicking copies the, now hidden, code to the clipboard, and double clicking switches a code between being shown and being hidden for the rare cases where you really do need to see it rather than just being able to paste it somewhere.

**Reply by semyl** [reply]
Thanks, I did find the option to hide the codes and that helps.

**Comment by fleish** [comment]
I would like to see this feature in all desktop versions. I only recently learned that the mobile app has the option already so it's just the desktop app where it is missing.

Locking in idle timeout _and_ when the computer is locked similar to what 1Password does would be ideal
